src/main/splitting/task_checker.py

```diff
@@ -0,0 +1,144 @@
+    @property
+    @abstractmethod
+    def language(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def min_symbols_number(self):
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def output_strings(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def create_source_file(self, source_code: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def is_source_file_correct(self, source_file: str):
+        raise NotImplementedError
+
+    @abstractmethod
+    def run_test(self, input: str, expected_output: str, source_file: str):
+        raise NotImplementedError
+
+    @staticmethod
+    def get_no_need_to_run_tests_values(rate: float, tasks_len: int):
+        need_to_run_tests = False
+        test_results = [rate] * tasks_len
+        return need_to_run_tests, test_results
+
+    def check_before_tests(self, source_file: str, source_code: str, tasks: list):
+        test_results = []
+        need_to_run_tests = True
+        rate = consts.TEST_RESULT.CORRECT_CODE.value
+
+        # not to check incorrect fragments
+        if not self.is_source_file_correct(source_file):
+            rate = consts.TEST_RESULT.INCORRECT_CODE.value
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        # not to check too small fragments because they cannot return true anyway
+        elif len(source_code) < self.min_symbols_number:
+            log.info('Code fragment is too small')
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        # not to check fragments without output because they cannot return anything
+        elif not does_string_contain_any_of_substrings(source_code, self.output_strings):
+            log.info('Code fragment does not contain any output strings')
+            need_to_run_tests, test_results = self.get_no_need_to_run_tests_values(rate, len(tasks))
+
+        return need_to_run_tests, test_results, rate
+
+    def check_task(self, task: str, in_and_out_files_dict: dict, source_file: str, stop_after_first_false=True):
+        log.info(f'Start checking task {task}')
+        in_and_out_files = in_and_out_files_dict.get(task)
+        if in_and_out_files is None:
+            log.error(f'Task data for the {task} does not exist')
+            raise ValueError(f'Task data for the {task} does not exist')
+
+        counted_tests, passed_tests = len(in_and_out_files), 0
+        for in_file, out_file in in_and_out_files:
+            is_passed = self.run_test(get_content_from_file(in_file), get_content_from_file(out_file), source_file)
+            log.info(f'Test {in_file} for task {task} is passed: {str(is_passed)}')
+            if is_passed:
+                passed_tests += 1
+            elif stop_after_first_false:
+                log.info('Stop after first false')
+                break
```

Comment:
Should we make the variable passed_tests = 0 before the break? Otherwise, we will have an incorrect result, because it can be passed some tests, passed_tests will be not zero and our rate will be not zero, but maybe other tests would be passed and the right value of rate must have to other value. Or I don't fully understand fully this logic